unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1775,6 +1786,75 @@ handleGist :: GistInput -> Cli ()
 handleGist (GistInput repo) =
   doPushRemoteBranch (GistyPush repo) Path.relativeEmpty' SyncMode.ShortCircuit
 
+handleDiffNamespaceToPatch :: Text -> DiffNamespaceToPatchInput -> Cli ()
+handleDiffNamespaceToPatch description input = do
+  Cli.Env {codebase} <- ask
+
+  branch1 <- Branch.head <$> Cli.resolveBranchId (input ^. #branchId1)
+  branch2 <- Branch.head <$> Cli.resolveBranchId (input ^. #branchId2)
+  branchDiff <- liftIO (BranchDiff.diff0 branch1 branch2)
```

Comment:
Did you consider using the V2 Branch diff instead? I think that's the way we want to be heading so we can avoid adding any more dependencies on the deepTerms/deepTypes stuff.
It doesn't currently diff metadata or patches, but I don't think you need that here do you?

You'd need to add an option to filter out `lib`, but I think we'll want that change there for future features anyways and it shouldn't be too hard to add.

Reply:
What is the V2 Branch diff? I just looked at how the `namespace.diff` command is implemented, and used some of those helpers.

Reply:
I re-implemented diffing on V2 branches for the Share surgical name index updates (since Share can't efficiently just load the whole V1 branch), you can find it here: https://github.com/unisonweb/unison/blob/trunk/parser-typechecker/src/U/Codebase/Branch/Diff.hs

Not only does it avoid loading the whole branch into memory at once, it also culls subtrees based on matching hashes, so in cases where the diff only exists in specific subtrees it won't even bother loading branches for other unaffected subtrees.